File: 1456. Maximum Number of Vowels in a Substring of Given Length/Solution.cpp
class Solution {
public:
    int maxVowels(string s, int k) {
        unordered_set<char> vowels;
        vowels.insert('a');
        vowels.insert('e');
        vowels.insert('o');
        vowels.insert('i');
        vowels.insert('u');
        int count = 0, res = 0, n = s.length();
        for (int i = 0; i < k; i++) {
            if (vowels.find(s[i]) != vowels.end()) count++;
        }
        res = count;
        for (int i = k, j = 0; i < n; i++, j++) {
            if (vowels.find(s[j]) != vowels.end()) count--;
            if (vowels.find(s[i]) != vowels.end()) count++;
            res = max(res, count);
        }
        return res;
    }
};

File: 1456. Maximum Number of Vowels in a Substring of Given Length/Solution.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vowels = set(['a', 'e', 'i', 'o', 'u'])
        count, res, n, left = 0, 0, len(s), 0
        for i in range(k):
            if s[i] in vowels: count += 1
        res = count
        for i in range(k, n):
            if s[left] in vowels: count -= 1
            if s[i] in vowels: count += 1
            left += 1
            res = max(res, count)
        return res